spec/integration/modules/write_file_spec.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'bolt_spec/conn'
+require 'bolt_spec/files'
+require 'bolt_spec/integration'
+require 'bolt_spec/run'
+
+describe 'running a plan with write_file' do
+  include BoltSpec::Conn
+  include BoltSpec::Files
+  include BoltSpec::Integration
+  include BoltSpec::Run
+
+  let(:filename) { 'hello.txt' }
+  let(:content) { 'Hello, world!' }
+  let(:params) { %W[target=#{transport} content=#{content} destination=#{filename}] }
+  let(:modulepath) { File.expand_path(File.join(__dir__, '../../fixtures/modules')) }
+  let(:inventory) { conn_inventory.merge(config) }
+  let(:config) do
+    {
+      'config' => {
+        'ssh' => { 'host-key-check' => false },
+        'winrm' => { 'ssl' => false }
+      }
+    }
+  end
+
+  shared_examples 'when writing a file from a plan' do
+    # Delete the uploaded file after each test
+    after(:each) do
+      result = run_command("#{remove_cmd} #{filename}", transport.to_s, inventory: inventory)
+      expect(result.first['status']).to eq('success')
+    end
+
+    it 'writes a file to the destination' do
+      with_tempfile_containing('inventory', YAML.dump(inventory), '.yaml') do |inv|
+        # Check that a file is correctly uploaded
+        result = run_cli_json(%W[plan run write_file -i #{inv.path} -m #{modulepath}] + params)
+
+        expect(result.size).to eq(1)
+        data = result.first
+        expect(data['status']).to eq('success')
+        expect(data['value']['_output']).to match(/Uploaded .* to .*hello.txt/)
+
+        # Check that the created file has the correct content
+        result = run_command("#{print_cmd} #{filename}", transport.to_s, inventory: inventory)
+
+        expect(result.size).to eq(1)
+        data = result.first
+        expect(data['status']).to eq('success')
+        expect(data['value']['stdout']).to match(/#{content}/)
+      end
+    end
+
+    it 'reports multiple function calls to analytics' do
```

Comment:
I originally had this in the unit tests, but kept getting an error that `file::write` and `upload_file` weren't functions. I'm not entirely sure why this was happening, but I'm thinking the environment the unit tests run in don't load all the functions?